### PR TITLE
3d Composability

### PR DESCRIPTION
**Pull request by mori360**
pytorch (fsdp, tp, pp) -> pytorch (composable)
Move (fsdp, tp, pp) tests under pytorch into a composable folder

TP:
test/distributed/tensor/parallel/test_ddp_2d_parallel.py
test/distributed/tensor/parallel/test_fsdp_2d_parallel.py
PP:
test/distributed/pipelining/test_composability.py
FSDP:
test/distributed/_composable/fsdp/test_fully_shard_trainin.py
-TestFullyShard2DTraining
-TestFullyShardHSDPTraining

=>
distributed/_composable/test_composability/test_noncontinuous.py
distributed/_composable/test_composability/test_continuous.py

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: test/distributed/_composable/test_composability/test_continuous.py [hunk 1]
```diff
@@ -48,8 +60,10 @@ def setUpClass(cls):
         Class-scope test fixture. Run once for entire test class, before any test starts.
         Set up the device.
         """
+        print(cls.rank)
```
Comment (weifengpy) [inline test/distributed/_composable/test_composability/test_continuous.py:63]: remove print ?

File: test/distributed/_composable/test_composability/test_continuous.py [hunk 2]
```diff
@@ -48,8 +60,10 @@ def setUpClass(cls):
...
         super().setUpClass()
         dev_id = cls.rank % torch.cuda.device_count()
+        print(f"cuda:{dev_id}")
```
Comment (weifengpy) [inline test/distributed/_composable/test_composability/test_continuous.py:66]: ditto

File: test/distributed/_composable/test_composability/test_continuous.py [hunk 3]
```diff
@@ -215,14 +229,14 @@ def build_stage(stage_idx, num_stages):
...
     world_size = int(os.getenv("WORLD_SIZE", 4))
-
     if rank != -1:
         # Launched with torchrun or other multi-proc launchers. Directly run the test.
         ComposabilityTest.run_rank(rank, world_size)
     else:
         # Launched as a single process. Spawn subprocess to run the tests.
         # Also need a rendezvous file for `init_process_group` purpose.
         rdvz_file = tempfile.NamedTemporaryFile(delete=False).name
+        print(rank, world_size)
```
Comment (weifengpy) [inline test/distributed/_composable/test_composability/test_continuous.py:239]: .

File: .ci/pytorch/multigpu-test.sh [hunk 4]
```diff
@@ -43,16 +43,15 @@ time python test/run_test.py --verbose -i distributed/_tensor/test_dtensor_compi
...
 time python test/run_test.py --verbose -i distributed/tensor/parallel/test_tp_random_state
 
 # FSDP2 tests
 time python test/run_test.py --verbose -i distributed/_composable/fsdp/test_fully_shard_training -- -k test_2d_mlp_with_nd_mesh
 
-# Pipelining composability tests
-time python test/run_test.py --verbose -i distributed/pipelining/test_composability.py
+# 3D composability tests
+time python test/run_test.py --verbose -i distributed/_composable/test_composability/test_noncontinuous.py
+time python test/run_test.py --verbose -i distributed/_composable/test_composability/test_continuous.py
```
Comment (weifengpy) [inline .ci/pytorch/multigpu-test.sh:54]: could I know why they are new tests are called `test_noncontinuous` and `test_continuous` ?

**Comment by weifengpy** [comment]
it might be more actionable if the PR is break down into 4 PRs. Are you interested in leanring ghstack for this purpose? If it's too much trouble, we can discuss if we need to collect stamps from each parallelsim before landing
* moving FSDP tests 
* moving TP tests 
* moving PP tests 
* moving DDP tests

File: .ci/pytorch/multigpu-test.sh (hunk 4 above)
Comment (mori360) [reply .ci/pytorch/multigpu-test.sh:54]: The `MultiProcContinousTest` at `test_continuous` relates to `MultiProcContinousTest`, thus name the test file as `test_continuous`.
The others tests relate the old style test. `test_noncontinuous` is not suitable, what do you think of `test_composability_fsdp_tp`

**Comment by Dawnrena01** [comment]
Could u possible help me with this 3d face app

**Pull request closed without merging**